test-lib.sh

```diff
@@ -285,6 +285,32 @@ function ct_npm_works() {
   : "  Success!"
 }
 
+# ct_binary_found_from_df binary [path]
+# --------------------
+# Checks if a binary can be found in PATH during Dockerfile build
+# Argument: binary - name of the binary to test accessibility for
+# Argument: path - optional path in which the binary should reside in
+#                  /opt/rh by default
+function ct_binary_found_from_df() {
+  local tmpdir
+  local binary=$1; shift
+  local binary_path=${1:-"/opt/rh"}
+  tmpdir=$(mktemp -d)
+  : "  Testing $binary in build from Dockerfile"
+
+  # Create Dockerfile that looks for the binary
+  cat <<EOF >"$tmpdir/Dockerfile"
+FROM $IMAGE_NAME
+RUN which $binary
+EOF
+  # Build an image, looking for expected path in the output
+  if ! docker build -f "$tmpdir/Dockerfile" --no-cache "$tmpdir" 2>&1 | grep "$binary_path"; then
```

Comment:
Is it enough to check only `/opt/rh`? What about to add as a parameter also `version` and check `full path` like was before in `grep -E`. To be more precise.

Reply:
You can already do that via the second argument, `/opt/rh` is only the default.

Reply:
What about `grep "^$binary_path"`?

Reply:
I am still thinking if it makes sense to somehow automatically "calculate" full binary path for a binary but the effort seems to be really high then expectation. Let's leave it to `^$binary_path'.

Reply:
I would leave that up to the user of the function. I have added the `^` to the default value though. Thanks for the idea!

Reply:
Thanks. LGTM.